client/src/auth/index.jsx

```diff
@@ -7,16 +7,18 @@ const PrivateRoute = ({ component: Component, ...rest }) => {
     <Route
       {...rest}
       component={props => {
-        return isAuth() ? (
-          <Component {...props} />
+        return isAuth ? (
+          <>
+            <Component {...props} />
+          </>
         ) : (
-            <>
-              {rest.clearUserInfo()}
-              <Redirect 
-                to={{ pathname: '/login', state: { from: props.location } }}
-              />
-            </>
-          );
+          <>
+            {rest.clearUserInfo()}
```

Comment:
@AymanAlqoqa I know you just moved this line, you didn't add it in this PR, but just for future knowledge: It would be better if this function call (`rest.clearUserInfo()`) was in componentDidMount instead of render.  It's best if render() has no function calls or side effects, since it can sometimes be called quite a lot.  Here, since there are brackets around the function call, react will actually try to put the function's return value into the dom (so you might see "undefined" or "null" or "false" appear in this component right before the redirect occurs).
Just something to keep in mind for the future :)

Reply:
ok, thanks